code/modules/hydroponics/grown.dm

```diff
@@ -128,6 +128,15 @@
 			if(seed.get_gene(/datum/plant_gene/trait/squash))
 				squash(hit_atom)
 
+/obj/item/food/grown/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(seed)
+		for(var/datum/plant_gene/trait/T in seed.genes)
+			T.on_attack(src, target, user, proximity)
+		if(seed.get_gene(/datum/plant_gene/trait/squash))
+			user.visible_message("<span class='warning'>[user] splats [src] against [target].</span>")
+			squash(target)
```

Comment:
why not put the liquid contents action in `on_attack` since you added it as a proc to genes in general?

Reply:
it was done for consistency with how throw_impact does it but now that you point it out it's pretty weird

Reply:
i'd assume it was done so liquid contents doesn't delete the plant before all the traits do their effects but I feel like traits that share those effects should be incompatible anyways